lib/flight_job/models/job.rb

```diff
@@ -179,7 +179,7 @@ def created_at
     end
 
     def results_dir
-      metadata['results_dir']
+      self.controls_file("results_dir").read || metadata['results_dir']
```

Comment:
We typically don't add `self.` to method calls unless there is a good reason to do so.  I don't see any reason why it needs to be here.  Can you remove it please.